.github/workflows/ci.yml

```diff
@@ -0,0 +1,103 @@
+name: CI
+
+env: 
+  IMAGE_NAME: activemq-artemis-self-provisioning-plugin
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript"]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout the repo
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.6.2
+
+      - name: Get node version
+        id: node
+        run: |
+            echo ":: set-output name=version::$(node -v)"
+
+      - name: Get node_modules cache
+        uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: |
+              ~/.yarn
+              **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+
+      - run: yarn install --frozen-lockfile
+
+      - name: Check for uncommited changes
+        run: git diff --quiet --exit-code
+      
+      # TODO: uncomment after adding tests
+      # - name: Execute tests
+      #   run: yarn run test
+
+      # TODO: uncomment after fixing storybook issue: https://github.com/openshift/dynamic-plugin-sdk/issues/216
+      # - name: Build storybook
+      #   run: |
+      #     yarn run build-storybook -- --quiet --loglevel silent
+
+      # - name: Deploy story book to github pages 🚀
+      #   uses: JamesIves/github-pages-deploy-action@v4.4.0
+      #   with:
+      #     branch: gh-pages # The branch the action should deploy to.
+      #     folder: storybook-static # The folder the action should deploy.
+
+      - name: Build the docker image
+        if: ${{ github.event_name == 'push' }}
```

Comment:
```suggestion

```
We don't need this check as we need to make sure docker image build is working fine on PR. Below push check is fine for push the image.